lib/analyzer/applications/java.ts

```diff
@@ -0,0 +1,56 @@
+import { JarFingerprint } from "../types";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+function mapContentToJarFingerprints(input: {
+  [fileName: string]: string;
+}): {
+  [path: string]: JarFingerprint[];
+} {
+  const jarFingerprints: JarFingerprint[] = Object.entries(input).map(
+    ([key, value]) => {
+      const path = key.substring(0, key.lastIndexOf("/"));
+      return {
+        location: path,
+        digest: value,
+      };
+    },
+  );
+
+  const resultAggregatedByPath = jarFingerprints.reduce(
+    (accumulator, current) => {
+      accumulator[current.location] = accumulator[current.location] || [];
+      accumulator[current.location].push(current);
+      return accumulator;
+    },
+    {},
+  );
+
+  return resultAggregatedByPath;
+}
+
+export async function jarFilesToScannedProjects(
+  filePathToContent: FilePathToContent,
+): Promise<AppDepsScanResultWithoutTarget[]> {
+  const mappedResult = mapContentToJarFingerprints(filePathToContent);
+  const scanResults: AppDepsScanResultWithoutTarget[] = [];
+
+  for (const key in mappedResult) {
+    if (Object.prototype.hasOwnProperty.call(mappedResult, key)) {
+      const element = mappedResult[key];
+      scanResults.push({
+        facts: [
+          {
+            type: "jarFingerprints",
+            data: element,
+          },
+        ],
+        identity: {
+          type: "maven",
```

Comment:
where is this used?

Reply:
I think this is then used in the registry to identify the package manager, but I'm not sure, will confirm with @ivanstanev tomorrow